### PR TITLE
feat: GLM-5.3-Flash family — runner, glm5 dialect, resident experts, batched prefill

### PR DESCRIPTION
**Pull request by NeelM0906**
## Summary

GLM-5.3-Flash (320B-A18B, 45 layers of Kimi Delta Attention and NoPE latent sparse attention over a four-stream mHC residual, 288 experts top-8) as a new Mference family, from PipeNetwork's mixed 4/8-bit MLX conversion (~181 GB installed). Family gate green on the 256 GB M3 Ultra; docs/families/GLM53_FLASH.md is the dossier.

- **Day-0 contract**: `glm53Flash` baseline, manifest validation of the new axes (`kvLoraRank`, `indexKPool`, `kdaGateLowerBound`, …), repack planner layout for the pre-quantized conversion, FAMILY_CONTRACT rows.
- **Toy parity**: goldens cut from PipeNetwork's parity-fixed runtime (16 experts top-8 so the toy exercises the production INT4 reduce); an fp32 Swift oracle reproduces every layer at 1e-4 with every discrete decision exact; `Glm53ForwardRunner` matches the oracle at the FP16 tier with no decision flips and 17/17 greedy tokens on both prompts.
- **Runner**: per-token decode with one command stream per token, GPU router (`glm53_router_select_k8_par`), KDA decode / pooled indexer / latent attention / mHC kernels (original transcriptions; nothing imported from ds4), INT4 experts with the swiglu clamp behind the existing function constant (other families' pipelines unchanged).
- **Resident experts**: the 171 GB set lives in per-layer anonymous shared buffers filled by `F_NOCACHE` reads (`.copied` strategy, GLM only; `.mapped` stays for every other family) and pinned in an `MTLResidencySet`; the routed FFN resolves expert offsets on the GPU with no CPU round trip. `auto` picks resident when pool + core + reserve fit physical memory.
- **Batched prefill**: 256-token chunks with matrix-unit fp32 INT8 GEMMs, a register-resident KDA chunk kernel (bit-identical to decode steps), grouped experts (511 GB/s on the real route distribution), dense causal attention below `index_topk` and per-query indexer selection past it. Per-token path stays one switch away (`MFERENCE_GLM53_BATCHED_PREFILL=0`).
- **Tokenizer**: `ChatDialect.glm5`, 22 HF renders byte-identical, `Glm5ToolCallParser`, the three `generation_config` EOS ids, `MFERENCE_GLM5_REASONING_EFFORT` knob.
- **Perf findings recorded**: the mmap resident strategy faults at 0.7 GB/s at this size; `dsv4_hc_weights` cost 52 of 86 ms per token (replaced by `glm53_hc_dots` / `glm53_hc_finalize`).

## Measured results (release CLI, `auto` = resident, protocol at `MFERENCE_GLM5_REASONING_EFFORT=low`)

| Case | Prompt / generated | Prefill | Decode | Peak RSS |
|---|---|---:|---:|---:|
| short-explanation | 60 / 653 | 1.25 s | 25.79 tok/s | 160 GiB |
| medium-review | 420 / 841 | 4.16 s (101 tok/s) | 24.30 tok/s | 160 GiB |
| long-synthesis | 2,792 / 709 | 33.28 s (84 tok/s) | 19.16 tok/s | 160 GiB |

9/9 measured footers `stop=endOfTurn`. Stated deviation: at the template's default `Max` effort the model does not leave its think block inside the 1,024-token cap (2,400 tokens of thinking measured), so the protocol ran at `low`, the template's own `reasoning_effort` value.

## Test plan

- [x] Full suite three times consecutively: 1,184 tests / 206 suites, each pass green (one pre-existing known issue also on `main`)
- [x] `swift build -c release` clean; `git diff --check` and `Scripts/check_markdown_links.rb` clean
- [x] `bringup-check.sh glm53flash scratch/glm53flash.gturbo` PASS (toy suite, `--verify-install` 49 files, 16 / 32 / auto ladder byte-identical)
- [x] First light: raw and chat probes coherent, dense A/B 32/32 identical, batched-vs-per-token prefill A/B agrees on the next-token argmax on all three protocol cases
- [x] Protocol run twice (per-token prefill and batched prefill), `MFERENCE_PHASES=1` snapshot recorded in the dossier
- [ ] Prefill is at roughly half the 150 tok/s lower target on longer prompts; the next levers (multi-query latent attention, GEMM instruction floor, grouped experts) are listed in the dossier

### CODE REVIEW

File: Sources/Mference/Runtime/Inference/Glm53ForwardRunner.swift [hunk 1]
```diff
@@ -0,0 +1,1142 @@
...
+
+        let tail = try open()
+        kernels.encodeHCCollapse(commandBuffer: tail, streams: streams, pre: meanPre, x: hiddenBuf,
+                                 hcMult: hc, hidden: hidden)
+        rms.encodeBF16W(commandBuffer: tail, x: hiddenBuf,
+                        weight: finalNorm.buffer, weightOffset: Int(finalNorm.offset),
+                        out: normed, d: UInt32(hidden), eps: eps)
+        try captureHalf("final_norm_out", normed, count: hidden)
+        if let logits {
+            gemvInt8(try open(), lmHead, x: normed, y: logits, m: cfg.vocabSize, n: hidden)
```
[the remark this answers is not in the document]
Comment (NeelM0906) [reply Sources/Mference/Runtime/Inference/Glm53ForwardRunner.swift]: Done in ab14584: the baseline and repack ArchInfo record unpaddedVocabSize 154,856 (154,820 BPE + 36 added ids; new manifests carry it), and both heads (decode and batched prefill) run glm53_mask_logits_tail so logits 154,856…154,879 are -inf before any sampler sees them. ManifestReader treats an absent field as unrecorded so existing installs still load; a kernel test covers the mask.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
